### PR TITLE
nodejs: fix impure configure phase

### PR DESCRIPTION
**Pull request by tie**
## Description of changes

When running configurePhase from impure build environment (i.e. nix develop), default ./configure script was selecting wrong, impure Python installation instead of native build inputs.

https://github.com/nodejs/node/blob/d57af10d1be390f4e33e6efe596133de9da36c86/configure#L7-L15

Split from #262124

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [x] aarch64-linux → x86_64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by aduh95 — APPROVED** [review]

File: pkgs/development/web/nodejs/nodejs.nix [hunk 1]
```diff
@@ -102,9 +102,14 @@ let
...
     ]) ++ extraConfigFlags;
 
-    configurePlatforms = [];
+    configurePlatforms = [ ];
 
-    dontDisableStatic = true;
+    configurePhase = ''
+      runHook preConfigure
+      CC_host="$CC_FOR_BUILD" CXX_host="$CXX_FOR_BUILD" \
+        ${python.executable} configure.py --prefix="$prefix" "''${configureFlags[@]}"
```
Comment (hsjobeki) [inline pkgs/development/web/nodejs/nodejs.nix:110]: I'm not a bash expert, but whats that `''${configureFlags[@]}` about ?
```suggestion
        ${python.executable} configure.py --prefix="$prefix" "${configureFlags[@]}"
```

Shouldn't it work without these empty quotes ?

Comment (tie) [reply pkgs/development/web/nodejs/nodejs.nix:110]: That’s Nix, not Bash 😅
Double quotes escape interpolation in multiline string, otherwise `configureFlags[@]` would be interpreted as a Nix expression.

Comment (tie) [reply pkgs/development/web/nodejs/nodejs.nix:110]: Hm, I wanted to post a link to the documentation but it looks like it’s not currently documented though IIRC it was before…
https://nix.dev/manual/nix/2.18/language/string-interpolation

Anyway, https://releases.nixos.org/nix-dev/2015-December/019018.html

Comment (hsjobeki) [reply pkgs/development/web/nodejs/nodejs.nix:110]: Oh right. The weird escaping of `$`

`{s = '' ''${} ''; }`

File: pkgs/development/web/nodejs/nodejs.nix [hunk 2]
```diff
@@ -54,14 +54,14 @@ let
 
     strictDeps = true;
 
+    __structuredAttrs = true;
```
Comment (drupol) [inline pkgs/development/web/nodejs/nodejs.nix:57]: why this change ?

Comment (tie) [reply pkgs/development/web/nodejs/nodejs.nix:57]: To avoid special cases for structured attributes by having them always enabled. Now that I think about it, probably makes more sense to use `configureScript` as a wrapper instead of overriding `configurePhase`.

Comment (tie) [reply pkgs/development/web/nodejs/nodejs.nix]: Updated the PR to use `configureScript`.

**Review by drupol — APPROVED** [review]

**Review by aduh95 — APPROVED** [review]

**Comment by tie** [comment]
@drupol, is this good for merge?
